### PR TITLE
[SPARK-40810][SQL] Use SparkIllegalArgumentException instead of IllegalArgumentException in CreateDatabaseCommand & AlterDatabaseSetLocationCommand

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
This pr aims to use SparkIllegalArgumentException instead of IllegalArgumentException in CreateDatabaseCommand & AlterDatabaseSetLocationCommand.

### Why are the changes needed?
When I work on https://issues.apache.org/jira/browse/SPARK-40790,
I found when `location` is empty, DDL command(CreateDatabaseCommand & AlterDatabaseSetLocationCommand) throw IllegalArgumentException, it seem not to fit into the new error framework.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Existed UT.

### CODE REVIEW

**Comment by panbingkun** [comment]
cc @MaxGekk

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/CreateNamespaceSuiteBase.scala [hunk 1]
```diff
@@ -68,10 +69,10 @@ trait CreateNamespaceSuiteBase extends QueryTest with DDLCommandTestUtils {
         val path = tmpDir.getCanonicalPath
         assert(!path.startsWith("file:/"))
 
-        val e = intercept[IllegalArgumentException] {
+        val e = intercept[SparkIllegalArgumentException] {
           sql(s"CREATE NAMESPACE $ns LOCATION ''")
         }
-        assert(e.getMessage.contains("Can not create a Path from an empty string"))
+        assert(e.getMessage.contains("Unsupported empty location"))
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/CreateNamespaceSuiteBase.scala:75]: As you are here, could you use `checkError()`.

Comment (panbingkun) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/command/CreateNamespaceSuiteBase.scala:75]: ok, let me do it.

File: core/src/main/resources/error/error-classes.json [hunk 2]
```diff
@@ -4207,5 +4207,10 @@
     "message" : [
       "Not enough memory to build and broadcast the table to all worker nodes. As a workaround, you can either disable broadcast by setting <autoBroadcastjoinThreshold> to -1 or increase the spark driver memory by setting <driverMemory> to a higher value<analyzeTblMsg>"
     ]
+  },
+  "_LEGACY_ERROR_TEMP_2251" : {
```
[the remark this answers is not in the document]
Comment (panbingkun) [reply core/src/main/resources/error/error-classes.json:4211]: ok

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @panbingkun.

File: core/src/main/resources/error/error-classes.json [hunk 3]
```diff
@@ -709,6 +709,11 @@
       }
     }
   },
+  "UNSUPPORTED_EMPTY_LOCATION" : {
+    "message" : [
+      "Unsupported empty location."
```
Comment (cloud-fan) [inline core/src/main/resources/error/error-classes.json:714]: empty location is a bit misleading. This is not a location with no files/directories, it's empty string. How about
```
EMPTY_STRING_AS_LOCATION: Can not reference a location with an empty string.
```

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/AlterNamespaceSetLocationSuiteBase.scala [hunk 4]
```diff
@@ -45,10 +46,13 @@ trait AlterNamespaceSetLocationSuiteBase extends QueryTest with DDLCommandTestUt
     val ns = s"$catalog.$namespace"
     withNamespace(ns) {
       sql(s"CREATE NAMESPACE $ns")
-      val message = intercept[IllegalArgumentException] {
-        sql(s"ALTER NAMESPACE $ns SET LOCATION ''")
-      }.getMessage
-      assert(message.contains("Can not create a Path from an empty string"))
+      val sqlText = s"ALTER NAMESPACE $ns SET LOCATION ''"
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/AlterNamespaceSetLocationSuiteBase.scala:49]: I think we can move this test to the base suite now, as the behavior is consistent between v1 and v2 tables.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/CreateNamespaceSuiteBase.scala [hunk 5]
```diff
@@ -67,15 +68,15 @@ trait CreateNamespaceSuiteBase extends QueryTest with DDLCommandTestUtils {
         // The generated temp path is not qualified.
         val path = tmpDir.getCanonicalPath
         assert(!path.startsWith("file:/"))
-
-        val e = intercept[IllegalArgumentException] {
-          sql(s"CREATE NAMESPACE $ns LOCATION ''")
-        }
-        assert(e.getMessage.contains("Can not create a Path from an empty string"))
-
+        val sqlText = s"CREATE NAMESPACE $ns LOCATION ''"
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/CreateNamespaceSuiteBase.scala:71]: ditto

**Pull request closed without merging**